src/overlays/actors/ovl_Mir_Ray3/z_mir_ray3.c

```diff
@@ -28,39 +28,397 @@ const ActorInit Mir_Ray3_InitVars = {
+                ptr[i].unk_0C.mf[0][0] = ptr[i].unk_0C.mf[1][1] = ptr[i].unk_0C.mf[2][2] = ptr[i].unk_0C.mf[3][3] =
+                    1.0f;
 
+                ptr[i].unk_0C.mf[0][1] = ptr[i].unk_0C.mf[0][2] = ptr[i].unk_0C.mf[0][3] = ptr[i].unk_0C.mf[1][0] =
+                    ptr[i].unk_0C.mf[1][2] = ptr[i].unk_0C.mf[1][3] = ptr[i].unk_0C.mf[2][0] = ptr[i].unk_0C.mf[2][1] =
+                        ptr[i].unk_0C.mf[2][3] = ptr[i].unk_0C.mf[3][0] = ptr[i].unk_0C.mf[3][1] =
+                            ptr[i].unk_0C.mf[3][2] = 0.0f;
+
+                if (Math3D_LineSegVsPlane(spEC[0], spEC[1], spEC[2], ptr[i].unk_4C->dist, &sp11C, &sp110, &sp104, 1)) {
+                    ptr[i].unk_0C.mf[0][0] = sp104.x - sp128.x;
+                    ptr[i].unk_0C.mf[0][1] = sp104.y - sp128.y;
+                    ptr[i].unk_0C.mf[0][2] = sp104.z - sp128.z;
+                }
+
+                sp11C.x = (shieldMf->mf[1][0] * 100.0f) + sp140.x;
+                sp11C.y = (shieldMf->mf[1][1] * 100.0f) + sp140.y;
+                sp11C.z = (shieldMf->mf[1][2] * 100.0f) + sp140.z;
+
+                sp110.x = (spF8[0] * 4.0f) + sp11C.x;
+                sp110.y = (spF8[1] * 4.0f) + sp11C.y;
+                sp110.z = (spF8[2] * 4.0f) + sp11C.z;
+
+                if (Math3D_LineSegVsPlane(spEC[0], spEC[1], spEC[2], ptr[i].unk_4C->dist, &sp11C, &sp110, &sp104, 1)) {
+                    ptr[i].unk_0C.mf[1][0] = sp104.x - sp128.x;
+                    ptr[i].unk_0C.mf[1][1] = sp104.y - sp128.y;
+                    ptr[i].unk_0C.mf[1][2] = sp104.z - sp128.z;
+                }
+            } else {
+                ptr[i].unk_4C = NULL;
+            }
+        }
+    }
+
+    spF8[0] = -(this->unk_260 * shieldMf->mf[2][0]);
+    spF8[1] = -(this->unk_260 * shieldMf->mf[2][1]);
+    spF8[2] = -(this->unk_260 * shieldMf->mf[2][2]);
+
+    sp134.x = (spF8[0] * (this->unk_214 * 400.0f)) + sp140.x;
+    sp134.y = (spF8[1] * (this->unk_214 * 400.0f)) + sp140.y;
+    sp134.z = (spF8[2] * (this->unk_214 * 400.0f)) + sp140.z;
+
+    if (!BgCheck_AnyLineTest1(&globalCtx->colCtx, &sp140, &sp134, &sp128, &spC4, 1)) {
+        Math_Vec3f_Copy(&sp128, &sp134);
+    }
+
+    sp128.x += spF8[0] * 5.0f;
+    sp128.y += spF8[1] * 5.0f;
+    sp128.z += spF8[2] * 5.0f;
+
+    spD4.x = (shieldMf->mf[0][0] * 300.0f) + sp140.x;
+    spD4.y = (shieldMf->mf[0][1] * 300.0f) + sp140.y;
+    spD4.z = (shieldMf->mf[0][2] * 300.0f) + sp140.z;
+
+    spC8.x = (shieldMf->mf[0][0] * 300.0f) + sp128.x;
+    spC8.y = (shieldMf->mf[0][1] * 300.0f) + sp128.y;
+    spC8.z = (shieldMf->mf[0][2] * 300.0f) + sp128.z;
+
+    Collider_SetQuadVertices(&this->colliderQuad, &spD4, &sp140, &spC8, &sp128);
+}
+#else
+void func_80B9E8D4(MirRay3* this, GlobalContext* globalCtx, MirRay3Struct* ptr);
 #pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Mir_Ray3/func_80B9E8D4.s")
+#endif
+
+#ifdef NON_MATCHING
+void MirRay3_Draw(Actor* thisx, GlobalContext* globalCtx) {
```

Comment:
Don't forget we can't have NON_MATCHING and NON_EQUIVALENT functions on overlays without breaking NON_MATCHING builds.
I suggest to change this to NON_EQUIVALENT and add a comment saying it actually is equivalent